File: v1/src/main/java/com/api/v1/car/domain/Car.java
package com.api.v1.car.domain;

import lombok.Getter;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;

import java.time.ZonedDateTime;
import java.util.UUID;

@Getter
@Document(collection = "v1_cars")
public class Car {

    @Id
    private UUID id;

    @Field
    private String make;

    @Field
    private String model;

    @Field
    private int productionYear;

    @Field
    private String vin;

    @Field
    private String plateNumber;

    @Field
    private String addedAt;

    @Field
    private String updatedAt;

    public Car(
            UUID id,
            String make,
            String model,
            int productionYear,
            String vin,
            String plateNumber,
            String addedAt
    ) {
        this.id = id;
        this.make = make;
        this.model = model;
        this.productionYear = productionYear;
        this.vin = vin;
        this.plateNumber = plateNumber;
        this.addedAt = addedAt;
    }

    public void update(
            String make,
            String model,
            int productionYear,
            String vin,
            String plateNumber
    ) {
        this.make = make;
        this.model = model;
        this.productionYear = productionYear;
        this.vin = vin;
        this.plateNumber = plateNumber;
        this.updatedAt = ZonedDateTime.now().toString();
    }

}

File: kotlin/v2/src/main/kotlin/com/api/v2/car/domain/Car.kt
package com.api.v2.car.domain

import com.api.v2.car.dtos.CarUpdateRequestDto
import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document
import java.time.ZonedDateTime
import java.util.UUID

@Document(collection = "v2_cars")
data class Car(
    @Id val id: UUID,
    val vin: String,
    var model: String,
    var productionYear: String,
    var make: String,
    var plateNumber: String,
    val createdAt: String,
    var updatedAt: String?
) {

    constructor(
        vin: String,
        model: String,
        productionYear: String,
        make: String,
        plateNumber: String,
    ): this(
        UUID.randomUUID(),
        vin,
        model,
        productionYear,
        make,
        plateNumber,
        ZonedDateTime.now().toString(),
        null
    )

    fun update(requestDto: CarUpdateRequestDto): Car {
        model = requestDto.model
        productionYear = requestDto.productionYear
        make = requestDto.make
        plateNumber = requestDto.plateNumber
        updatedAt = ZonedDateTime.now().toString()
        return this
    }

}